[BUG] test_mixed_compress_read orc_test.py failures

**Describe the bug**
our nightly integration tests failued:
```
08:42:31  FAILED ../../src/main/python/orc_test.py::test_mixed_compress_read[{'spark.rapids.sql.format.orc.reader.type': 'PERFILE'}-][IGNORE_ORDER]
08:42:31  FAILED ../../src/main/python/orc_test.py::test_mixed_compress_read[{'spark.rapids.sql.format.orc.reader.type': 'PERFILE'}-orc][IGNORE_ORDER]
08:42:31  FAILED ../../src/main/python/orc_test.py::test_mixed_compress_read[{'spark.rapids.sql.format.orc.reader.type': 'MULTITHREADED'}-][IGNORE_ORDER]
08:42:31  FAILED ../../src/main/python/orc_test.py::test_mixed_compress_read[{'spark.rapids.sql.format.orc.reader.type': 'MULTITHREADED'}-orc][IGNORE_ORDER]
08:42:31  FAILED ../../src/main/python/orc_test.py::test_mixed_compress_read[{'spark.rapids.sql.format.orc.reader.type': 'COALESCING'}-][IGNORE_ORDER]
08:42:31  FAILED ../../src/main/python/orc_test.py::test_mixed_compress_read[{'spark.rapids.sql.format.orc.reader.type': 'COALESCING'}-orc][IGNORE_ORDER]
```

```
08:42:31  [1m[31mE                   Caused by: java.lang.NullPointerException[0m
08:42:31  [1m[31mE                   	at com.nvidia.spark.rapids.shims.v2.OrcShims$.newOrcOutStream(OrcShims.scala:75)[0m
08:42:31  [1m[31mE                   	at com.nvidia.spark.rapids.OrcCodecWritingHelper.$anonfun$withCodecOutputStream$1(GpuOrcScanBase.scala:1539)[0m
08:42:31  [1m[31mE                   	at com.nvidia.spark.rapids.Arm.withResource(Arm.scala:28)[0m
08:42:31  [1m[31mE                   	at com.nvidia.spark.rapids.Arm.withResource$(Arm.scala:26)[0m
08:42:31  [1m[31mE                   	at com.nvidia.spark.rapids.OrcDataStripe.withResource(GpuOrcScanBase.scala:1555)[0m
08:42:31  [1m[31mE                   	at com.nvidia.spark.rapids.OrcCodecWritingHelper.withCodecOutputStream(GpuOrcScanBase.scala:1523)[0m
08:42:31  [1m[31mE                   	at com.nvidia.spark.rapids.OrcCodecWritingHelper.withCodecOutputStream$(GpuOrcScanBase.scala:1518)[0m
08:42:31  [1m[31mE                   	at com.nvidia.spark.rapids.OrcDataStripe.withCodecOutputStream(GpuOrcScanBase.scala:1555)[0m
08:42:31  [1m[31mE                   	at com.nvidia.spark.rapids.OrcDataStripe.$anonfun$stripeSize$2(GpuOrcScanBase.scala:1584)[0m
08:42:31  [1m[31mE                   	at com.nvidia.spark.rapids.OrcDataStripe.$anonfun$stripeSize$2$adapted(GpuOrcScanBase.scala:1583)[0m
08:42:31  [1m[31mE                   	at com.nvidia.spark.rapids.Arm.withResource(Arm.scala:28)[0m
08:42:31  [1m[31mE                   	at com.nvidia.spark.rapids.Arm.withResource$(Arm.scala:26)[0m
08:42:31  [1m[31mE                   	at com.nvidia.spark.rapids.OrcDataStripe.withResource(GpuOrcScanBase.scala:1555)[0m
08:42:31  [1m[31mE                   	at com.nvidia.spark.rapids.OrcDataStripe.$anonfun$stripeSize$1(GpuOrcScanBase.scala:1583)[0m
08:42:31  [1m[31mE                   	at com.nvidia.spark.rapids.OrcDataStripe.$anonfun$stripeSize$1$adapted(GpuOrcScanBase.scala:1582)[0m
08:42:31  [1m[31mE                   	at com.nvidia.spark.rapids.Arm.withResource(Arm.scala:28)[0m
08:42:31  [1m[31mE                   	at com.nvidia.spark.rapids.Arm.withResource$(Arm.scala:26)[0m
08:42:31  [1m[31mE                   	at com.nvidia.spark.rapids.OrcDataStripe.withResource(GpuOrcScanBase.scala:1555)[0m
08:42:31  [1m[31mE                   	at com.nvidia.spark.rapids.OrcDataStripe.stripeSize$lzycompute(GpuOrcScanBase.scala:1582)[0m
08:42:31  [1m[31mE                   	at com.nvidia.spark.rapids.OrcDataStripe.stripeSize(GpuOrcScanBase.scala:1569)[0m
08:42:31  [1m[31mE                   	at com.nvidia.spark.rapids.OrcDataStripe.getBlockSize(GpuOrcScanBase.scala:1565)[0m
08:42:31  [1m[31mE                   	at com.nvidia.spark.rapids.MultiFileOrcPartitionReader.$anonfun$calculateEstimatedBlocksOutputSize$2(GpuOrcScanBase.scala:1784)[0m
08:42:31  [1m[31mE                   	at com.nvidia.spark.rapids.MultiFileOrcPartitionReader.$anonfun$calculateEstimatedBlocksOutputSize$2$adapted(GpuOrcScanBase.scala:1782)[0m
08:42:31  [1m[31mE                   	at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)[0m
08:42:31  [1m[31mE                   	at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)[0m
08:42:31  [1m[31mE                   	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)[0m
08:42:31  [1m[31mE                   	at com.nvidia.spark.rapids.MultiFileOrcPartitionReader.$anonfun$calculateEstimatedBlocksOutputSize$1(GpuOrcScanBase.scala:1782)[0m
08:42:31  [1m[31mE                   	at com.nvidia.spark.rapids.MultiFileOrcPartitionReader.$anonfun$calculateEstimatedBlocksOutputSize$1$adapted(GpuOrcScanBase.scala:1776)[0m
08:42:31  [1m[31mE                   	at scala.collection.mutable.LinkedHashMap.foreach(LinkedHashMap.scala:152)[0m
08:42:31  [1m[31mE                   	at com.nvidia.spark.rapids.MultiFileOrcPartitionReader.calculateEstimatedBlocksOutputSize(GpuOrcScanBase.scala:1776)[0m
08:42:31  [1m[31mE                   	at com.nvidia.spark.rapids.MultiFileCoalescingPartitionReaderBase.$anonfun$readPartFiles$1(GpuMultiFileReader.scala:797)[0m
08:42:31  [1m[31mE                   	at com.nvidia.spark.rapids.Arm.withResource(Arm.scala:28)[0m
08:42:31  [1m[31mE                   	at com.nvidia.spark.rapids.Arm.withResource$(Arm.scala:26)[0m
08:42:31  [1m[31mE                   	at com.nvidia.spark.rapids.FilePartitionReaderBase.withResource(GpuMultiFileReader.scala:235)[0m
```

--- a/sql-plugin/src/main/320+/scala/com/nvidia/spark/rapids/shims/v2/OrcShims.scala
+++ b/sql-plugin/src/main/320+/scala/com/nvidia/spark/rapids/shims/v2/OrcShims.scala
@@ -72,7 +72,10 @@ object OrcShims {
   // create ORC out stream
   def newOrcOutStream(name: String, bufferSize: Int, codec: CompressionCodec,
       receiver: PhysicalWriter.OutputReceiver): OutStream = {
-    val options = new StreamOptions(bufferSize).withCodec(codec, codec.getDefaultOptions)
+    val options = new StreamOptions(bufferSize)
+    if (codec != null) {
+      options.withCodec(codec, codec.getDefaultOptions)
+    }
     new OutStream(name, options, receiver)
   }
 

